### PR TITLE
Close #474

### LINKED ISSUE #474: Linting OpenAPI document in Travis
**Is your feature request related to a problem? Please describe.**
I'd like to have the generated OpenAPI document formally correct and tested in our CI infrastructure. This would allow to not have a broken swagger page in our API endpoint.

**Describe the solution you'd like**
Given I didn't find a decent and well-maintained python package after a quick assessment I find out [spectral](https://github.com/stoplightio/spectral) as one of the best linter out there. It is released as open-source and available as npm package or docker image. I have opted to use the latter in Travis.

**Describe alternatives you've considered**
Other solutions seems not to be as robust as well.

**Additional context**

### PR DESCRIPTION
**Pull request by francbartoli**
Fix trailing slash in server url

Fix tag not declared in the root mapping key

Fix missing operationId

Add spectral command to travis for linting the openapi document

### CODE REVIEW

File: pygeoapi/openapi.py [hunk 1]
```diff
@@ -153,13 +153,14 @@ def get_oas_30(cfg):
             'summary': 'Landing page',
             'description': 'Landing page',
             'tags': ['server'],
+            'operationId': 'getRoot',
```
Comment (tomkralidis) [inline pygeoapi/openapi.py:156]: let's align the `operationId` names as much as possible with https://github.com/opengeospatial/ogcapi-features/blob/master/core/examples/openapi/ogcapi-features-1-example1.yaml ?

File: pygeoapi/provider/ogr.py [hunk 2]
```diff
@@ -427,6 +427,10 @@ def _load_source_helper(self, source_type):
 
     def _get_next_feature(self, layer, feature_id):
         try:
+            if layer.GetFeatureCount() == 0:
```
Comment (tomkralidis) [inline pygeoapi/provider/ogr.py:430]: Should changes in this file be in another PR?

File: tests/test_ogr_wfs_provider.py [hunk 3]
```diff
@@ -137,7 +137,7 @@ def config_geonode_gs_WFS():
         'data': {
             'source_type': 'WFS',
             'source':
-                'WFS:https://geonode.wfp.org/geoserver/wfs?',
```
Comment (tomkralidis) [inline tests/test_ogr_wfs_provider.py:140]: here too, should this be in another PR?

File: tests/test_ogr_csv_provider.py [hunk 4]
```diff
@@ -85,7 +85,7 @@ def test_get_vsicurl(config_vsicurl_csv):
     p = OGRProvider(config_vsicurl_csv)
     result = p.get('32')
     assert result['id'] == 32
-    assert '11' in result['properties']['codice_regione']
```
Comment (tomkralidis) [inline tests/test_ogr_csv_provider.py:88]: here too, should this be in another PR?

File: pygeoapi/provider/ogr.py (hunk 2 above)
Comment (francbartoli) [reply pygeoapi/provider/ogr.py:430]: They were not intended to be there, sorry. Also, those changes have been already merged

**Comment by francbartoli** [comment]
@tomkralidis I started a new clean PR #480 to resolve some of the above comments. It contains the alignment with the example. Can you have a look?
